### PR TITLE
Add support for USE_COUNTER2_* histograms.

### PR DESCRIPTION
**Pull request by vitillo**

### CODE REVIEW

**Comment by mreid-moz** [comment]
Is there a bug or some background info for USE_COUNTER2_* histograms?

**Comment by mreid-moz** [comment]
oh duh, bug 1204994

File: mozaggregator/aggregator.py [hunk 1]
```diff
@@ -125,7 +122,22 @@ def _extract_main_histograms(state, histograms, is_child):
     if type(histograms) != dict:
         return
 
+    # Deal with USE_COUNTER2_ histograms, see Bug 1204994
+    docs_destroyed = histograms.get("CONTENT_DOCUMENTS_DESTROYED", {}).get("sum", None)
+    top_docs_destroyed = histograms.get("TOP_LEVEL_CONTENT_DOCUMENTS_DESTROYED", {}).get("sum", None)
+
+    if docs_destroyed is not None and top_docs_destroyed is not None:
```
Comment (mreid-moz) [inline mozaggregator/aggregator.py]: Should probably check that these are numbers >= 0

File: mozaggregator/aggregator.py [hunk 2]
```diff
@@ -125,7 +122,22 @@ def _extract_main_histograms(state, histograms, is_child):
...
+        total_destroyed = docs_destroyed - top_docs_destroyed
```
Comment (mreid-moz) [inline mozaggregator/aggregator.py]: As discussed on IRC, I don't think you need to do this. You should track the count of "usage" and "non-usage" using one or the other of `docs_destroyed` or `top_docs_destroyed` based on the suffix of the histogram name (either `_DOCUMENT` or `_PAGE`)

File: mozaggregator/aggregator.py [hunk 3]
```diff
@@ -125,7 +122,22 @@ def _extract_main_histograms(state, histograms, is_child):
...
+    else:
+        total_destroyed = None
+
     for histogram_name, histogram in histograms.iteritems():
+        if total_destroyed is not None and histogram_name.startswith("USE_COUNTER2_"):
+            used = histogram.get("values", {}).get("1", None)
+            if used is None:
+                continue
+            histogram["values"]["0"] = total_destroyed - used
```
Comment (mreid-moz) [inline mozaggregator/aggregator.py]: Should be tracking the number of uses and the number of non-uses.

File: mozaggregator/aggregator.py [hunk 4]
```diff
@@ -125,7 +122,22 @@ def _extract_main_histograms(state, histograms, is_child):
     if type(histograms) != dict:
         return
 
+    # Deal with USE_COUNTER2_ histograms, see Bug 1204994
+    pages_destroyed = histograms.get("TOP_LEVEL_CONTENT_DOCUMENTS_DESTROYED", {}).get("sum", -1)
+    docs_destroyed = histograms.get("CONTENT_DOCUMENTS_DESTROYED", {}).get("sum", -1)
+
     for histogram_name, histogram in histograms.iteritems():
+        if pages_destroyed >= 0 and histogram_name.startswith("USE_COUNTER2_") and histogram_name.endswith("_PAGE"):
```
Comment (mreid-moz) [inline mozaggregator/aggregator.py]:
```
>>> "foo" > 0
True
```
:(

File: mozaggregator/aggregator.py [hunk 5]
```diff
@@ -125,7 +122,22 @@ def _extract_main_histograms(state, histograms, is_child):
...
+            used = histogram.get("values", {}).get("1", -1)
+            if used <= 0:
+                continue
+            histogram["values"]["0"] = pages_destroyed - used
```
Comment (mreid-moz) [inline mozaggregator/aggregator.py]: So `histogram["values"]["1"]` will be the count of pages in which the feature was used and `histogram["values"]["0"]` will be the count of pages where it was *not* used?

File: tests/test_db.py [hunk 6]
```diff
@@ -156,6 +150,9 @@ def test_submission_dates_metrics():
 
 @nottest
 def test_histogram(prefix, channel, version, dates, metric, value, expected_count):
+    if metric.endswith("CONTENT_DOCUMENTS_DESTROYED"):  # Ignore USE_COUNTER2_ support histograms
```
Comment (mreid-moz) [inline tests/test_db.py]: Shouldn't these histograms get aggregates as normal?

File: mozaggregator/aggregator.py (hunk 5 above)
Comment (vitillo) [inline mozaggregator/aggregator.py]: Yes, that should make sense given that the histogram is boolean.

File: tests/test_db.py (hunk 6 above)
Comment (mreid-moz) [inline tests/test_db.py]: Never mind, this is an artifact of how things are tested.

**Comment by mreid-moz** [comment]
Looks good now, r+

**Pull request merged**